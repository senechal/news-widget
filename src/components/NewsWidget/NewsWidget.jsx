import React from 'react';
import NewsItem from '../NewsItem';

const NewsWidget = () => (
  <div>
    News Widget Component
    <NewsItem />
  </div>
);

export default NewsWidget;
