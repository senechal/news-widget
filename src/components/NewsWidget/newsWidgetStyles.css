.news-widget-list {
  max-width: 95vw;
  width: 500px;
  margin: 0 auto;
  background-color: #fafafa;
  padding: 20px 10px;
}

.news-header {
  vertical-align: middle;
  margin-bottom: 30px;
}

.news-title {
  color: rgb(243, 118, 187);
  border-bottom: 2px solid rgb(243, 118, 187);
  display: inline;
}

.source-selector{
  display: inline;
  float: right;
  background-color: #fff;
  padding: 5px 20px;
  border: 2px solid #777;
  border-radius: 4px;
  color: #777;
}

.show-more-button{
  background-color: #fff;
  padding: 12px 20px;
  border: 2px solid rgb(243, 118, 187);
  border-radius: 4px;
  color: rgb(243, 118, 187);
  font-size: 16px;
  cursor: pointer;
  transition: all 200ms ease;
}

.show-more-button:hover{
  background-color: rgb(243, 118, 187);
  color: #fff;
}

.news-list{
  margin-bottom: 30px;
}