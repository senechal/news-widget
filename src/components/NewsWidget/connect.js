import { connect } from 'react-redux';
import NewsWidget from './NewsWidget';
import {
  fetchNews,
  fetchSources,
  fetchMore,
  filterBySource,
} from '../../actions/NewsWidgetActions';

const reducer = 'widget';

const mapStateToProps = state => ({ ...state[reducer] });
const mapDispatchToProps = {
  fetchNews,
  fetchSources,
  fetchMore,
  filterBySource,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NewsWidget);
