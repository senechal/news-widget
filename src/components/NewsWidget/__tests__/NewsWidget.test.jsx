import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import NewsWidget from '../NewsWidget';

describe('Test for NewsWidget', () => {
  configure({ adapter: new Adapter() });
  it('should render correctly', () => {
    const wrapper = shallow(<NewsWidget />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
