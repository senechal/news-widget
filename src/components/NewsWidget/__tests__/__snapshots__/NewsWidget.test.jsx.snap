// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NewsWidget should render correctly 1`] = `
<div>
  News Widget Component
  <NewsItem />
</div>
`;
